src/tests/tests.h

```diff
@@ -566,6 +566,10 @@ class Test {
 
       const std::optional<CodeLocation>& registration_location() const { return m_registration_location; }
 
+      void set_serialization(bool needs_serialization) { m_needs_serialization = needs_serialization; }
```

Comment:
This feels weird in two ways

- Serialization state can nominally change (eg a `Test` can call `set_serialization(true)` then `set_serialization(false)`)
- A `Test` can set serialization late (after initialization)

I think both are remedied by making it an argument to the (newly added) `Test::initialize`

Reply:
Agreed.